platform/mock-local-home-platform.ts

```diff
@@ -69,38 +108,56 @@ export class MockLocalHomePlatform implements MockUDPListener {
 
   private sendUDPBroadcast(scanConfig: UDPScanConfig) {
     const packetBuffer = Buffer.from(scanConfig.discoveryPacket, 'hex');
-    this.network.sendUDPMessage(
+    this.mockNetwork.sendUDPMessage(
       packetBuffer,
       scanConfig.broadcastPort,
       scanConfig.broadcastAddress,
       scanConfig.listenPort,
       scanConfig.broadcastAddress
     );
   }
+  private isPromise<RS>(response: RS | Promise<RS>): response is Promise<RS> {
+    return (response as Promise<RS>).then !== undefined;
+  }
+
+  private processIntentResponse<RS>(response: RS | Promise<RS>): Promise<RS> {
```

Comment:
If you mark this function as an `async`, you can `await` the `response` with [minimal side effects](https://stackoverflow.com/questions/55262996/does-awaiting-a-non-promise-have-any-detectable-effect) regardless of whether it is a Promise or not. It'll clean up this code.

Reply:
used [Proppy's solution](https://github.com/googleinterns/smart-home-local/pull/29#discussion_r447494848) with `async` and `await`